src/verify_jwk_rsa_pss_test.cc

```diff
@@ -0,0 +1,382 @@
+    "eyJhbGciOiJQUzI1NiIsInR5cCI6IkpXVCJ9."
+    "eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlh"
+    "dCI6MTUxNjIzOTAyMn0."
+    "hZnl5amPk_I3tb4O-Otci_5XZdVWhPlFyVRvcqSwnDo_srcysDvhhKOD01DigPK1lJvTSTol"
+    "yUgKGtpLqMfRDXQlekRsF4XhAjYZTmcynf-C-6wO5EI4wYewLNKFGGJzHAknMgotJFjDi_NC"
+    "VSjHsW3a10nTao1lB82FRS305T226Q0VqNVJVWhE4G0JQvi2TssRtCxYTqzXVt22iDKkXeZJ"
+    "ARZ1paXHGV5Kd1CljcZtkNZYIGcwnj65gvuCwohbkIxAnhZMJXCLaVvHqv9l-AAUV7esZvkQ"
+    "R1IpwBAiDQJh4qxPjFGylyXrHMqh5NlT_pWL2ZoULWTg_TJjMO9TuQ";
+
+
+const std::string JwtTextWithNonExistentKid =
+    "eyJhbGciOiJQUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6Im5vbmV4aXN0ZW50In0."
+    "eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlh"
+    "dCI6MTUxNjIzOTAyMn0."
+    "USMoL8XwVl-sqtIl-VQr97oNr1XWbgnJnDJbi65ExV7IioYQ3cGfrpi9n2GxJOwuw6zU572l"
+    "ME-wD9It-Q8H8eAOi83KoimQJmdzCGGUGTgwo3tZK5HV7W3srgP1_46-X43DYWOT6h1pIAE7"
+    "7s23XuSKbq4rpp6cmbDODARfTj6OTQWTqwhOkX0Xo7i2q1foreKI8PnOyrvbs7oXrLJGZhg_"
+    "6mRnP0wRJJFkIu2uYKcLDcgJ0OWXY6dQ-8agj-yjZ5ZUX8GUcy347P0UUpsGVNd1pUawLwTi"
+    "kmNidJOxkGlawLtOwE7u0WtZdYmcppx99Qw5U4gYdQQx0wJqgj_d8g";
+
+
+// Expected behavior for VerifyKidMatchingTest:
+// If kid is not specified in the jwt, allow verification as long as any of the
+//   keys in the jwks are appropriate.
+// If kid is specified in the jwt, use only the requested key in the jwks for
+//   verification.
+class VerifyKidMatchingTest : public testing::Test {
+ protected:
+  void SetUp() {
+    correct_jwks_ = Jwks::createFrom(JwtIoPublicKeyRSAPSS, Jwks::Type::JWKS);
+    EXPECT_EQ(correct_jwks_->getStatus(), Status::Ok);
+    wrong_jwks_ = Jwks::createFrom(PublicKeyRSAPSS, Jwks::Type::JWKS);
+    EXPECT_EQ(wrong_jwks_->getStatus(), Status::Ok);
+  }
+
+  // This jwks contains the appropriate key for signature verification
+  JwksPtr correct_jwks_;
+  // This jwks does not contain the appropriate key for signature verification
+  JwksPtr wrong_jwks_;
+};
+
+
+TEST_F(VerifyKidMatchingTest, JwtTextWithNoKidNoMatchingKey) {
+  Jwt jwt;
+  EXPECT_EQ(jwt.parseFromString(JwtTextWithNoKid), Status::Ok);
+  // jwt has no kid, and none of the keys in the jwks can be used to verify,
+  //   hence verification fails
+  EXPECT_EQ(verifyJwt(jwt, *wrong_jwks_),
+            Status::JwtVerificationFail);
+}
+
+
+TEST_F(VerifyKidMatchingTest, JwtTextWithNoKidOk) {
+  Jwt jwt;
+  EXPECT_EQ(jwt.parseFromString(JwtTextWithNoKid), Status::Ok);
+  // jwt has no kid, and one of the keys in the jwks can be used to verify,
+  //   hence verification is ok
+  EXPECT_EQ(verifyJwt(jwt, *correct_jwks_, 1), Status::Ok);
+}
+
+
+TEST_F(VerifyKidMatchingTest, JwtTextWithNonExistentKid) {
+  Jwt jwt;
+  EXPECT_EQ(jwt.parseFromString(JwtTextWithNonExistentKid), Status::Ok);
+  // jwt has a kid, which did not match any of the keys in the jwks (even
+  //   though the jwks does contain an appropriate key)
+  EXPECT_EQ(verifyJwt(jwt, *correct_jwks_, 1),
```

Comment:
Could you run 

script/check-style

to check the code format. 

It seems that these two line can be in one line.  
same as line 356

Reply:
Done! 2bafc33